src/resizing.jl

```diff
@@ -18,7 +18,10 @@ function restrict{T,N}(A::AbstractArray{T,N}, dim::Integer)
         return A
     end
     newsz = ntuple(i->i==dim?restrict_size(size(A,dim)):size(A,i), Val{N})
-    out = Array{typeof(A[1]/4+A[2]/2),N}(newsz)
+    # FIXME: The following line fails for interpolations because
```

Comment:
`first(A)` works for interpolations. `A[1]` does not. For extrapolations neither work